### PR TITLE
Bugfix for Applepay subtotal calculation

### PR DESCRIPTION
**Pull request by lancergr**
**Description**
Fix ApplePay subtotal calculation

**Tested scenarios**
ApplePay payment amount calculation

### CODE REVIEW

File: app/code/community/Adyen/Payment/Block/ApplePay.php [hunk 1]
```diff
@@ -62,13 +62,20 @@ public function getSubTotal()
...
+        } else if (Mage::getSingleton('checkout/session')->getQuote()->getItemsCount() > 0) {
+            $cart = Mage::getModel('checkout/cart');
+            $subtotal['label'] = $this->__('Grand Total');
+
+            // Get the amount including tax
+            $totals = $cart->getQuote()->getTotals();
+            if (method_exists($totals['subtotal'], "getValueInclTax")) {
+                $amount = $totals['subtotal']->getValueInclTax();
+            } else {
+                $amount = $totals['subtotal']->getValue() + $totals['tax']->getValue();
```
Comment (rikterbeek) [inline app/code/community/Adyen/Payment/Block/ApplePay.php]: Resulting in:
Fatal error: Call to a member function getValue() on a non-object in /var/www/htdocs/app/code/community/Adyen/Payment/Block/ApplePay.php on line 74

Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Block/ApplePay.php]: if you clear the basket/cache and add a new item you will get this error becuase the $totals['tax'] is not calculated yet. Trying to find out if we can set it but even with recalculating the totals with:
$quote->setTotalsCollectedFlag(false);
$quote->collectTotals();
does not solve the problem.

Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Block/ApplePay.php]: I think we need to calculate the tax if customer is logged in and has a default address. If not calculate the tax on the apple pay callback when we have retrieved the address + shopper detaills from the apple pay wallet.

File: app/code/community/Adyen/Payment/Block/ApplePay.php [hunk 2]
```diff
@@ -62,12 +62,35 @@ public function getSubTotal()
...
+    public function getSubtotalInclTax()
+    {
+        $cart = Mage::getModel('checkout/cart');
+        $subtotal = 0;
+        $totals = $cart->getQuote()->getTotals();
+        $config = Mage::getSingleton('tax/config');
+        if (isset($totals['subtotal'])) {
+            if ($config->displayCartSubtotalBoth()) {
+                $subtotal = $totals['subtotal']->getValueInclTax();
+            } elseif($config->displayCartSubtotalInclTax()) {
```
Comment (Aleffio) [inline app/code/community/Adyen/Payment/Block/ApplePay.php]: can we merge this condition with the one in Line 85? 
$config->displayCartSubtotalBoth() || $config->displayCartSubtotalInclTax()

Comment (lancergr) [reply app/code/community/Adyen/Payment/Block/ApplePay.php]: done

**Pull request merged**